into/backends/sparksql.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import division, print_function, absolute_import
+
+from collections import Iterator
+import pandas as pd
+import itertools
+import datashape
+from datashape import dshape, Record, DataShape, Option, Tuple
+from datashape.predicates import isdimension
+from toolz import valmap
+
+from ..convert import convert
+from ..append import append
+from ..core import discover
+from ..into import convert, append
+from .spark import RDD, SchemaRDD, Dummy
+
+
+try:
+    from pyspark.sql import SQLContext
+    from pyspark.sql import (ByteType, ShortType, IntegerType, LongType,
+                             FloatType, DoubleType, StringType, BinaryType,
+                             BooleanType, TimestampType, DateType, ArrayType,
+                             StructType, StructField, MapType)
+except ImportError:
+    SchemaRDD = SQLContext = Dummy
+    ByteType = ShortType = IntegerType = LongType = FloatType = Dummy
+    MapType = DoubleType = StringType = BinaryType = BooleanType = Dummy
+    TimestampType = DateType = StructType = ArrayType = StructField = Dummy
+
+
+@convert.register(pd.DataFrame, SchemaRDD)
+def sparksql_dataframe_to_pandas_dataframe(rdd, dshape=None, **kwargs):
+    return pd.DataFrame(convert(list, rdd, **kwargs),
+                        columns=dshape.measure.names)
+
+
+_names = ('tmp%d' % i for i in itertools.count())
+
+
+@append.register(SQLContext, (list, tuple, Iterator))
+def iterable_to_sql_context(ctx, seq, **kwargs):
+    return append(ctx, ctx._sc.parallelize(seq), **kwargs)
+
+
+@append.register(SQLContext, pd.DataFrame)
+def frame_to_sqlcontext(ctx, df, **kwargs):
+    return append(ctx, convert(Iterator, df, **kwargs), **kwargs)
+
+
+@append.register(SQLContext, RDD)
```

Comment:
Is there any way to make this a convert statement?  Does the context of the RDD know its SQLContext?

If we can get a convert route all the way to SchemaRDD then we'll be very happy because that will connect the local world to all of the things that SchemaRDD can connect to (like `HDFS(Parquet)`) in a single function call.